File: benchmark/go/sort_heavy.go
package main

import "fmt"

func main() {
	const n = 20000
	a := make([]int64, n)
	for p := int64(0); p < n; p++ {
		a[p] = ((p*1_664_525 + 1_013_904_223) % 2_147_483_647) % 100_000
	}

	// Insertion sort — same algorithm as all three language variants
	for i := 1; i < n; i++ {
		key := a[i]
		j := i
		for j > 0 && a[j-1] > key {
			a[j] = a[j-1]
			j--
		}
		a[j] = key
	}

	var checksum int64
	for q := 0; q < n; q++ {
		checksum += a[q] * int64(q+1)
	}
	fmt.Println(checksum)
}

File: benchmark/rust/sort_heavy.rs
fn main() {
    let n: usize = 20000;
    let mut a: Vec<i64> = (0..n as i64)
        .map(|p| ((p * 1_664_525 + 1_013_904_223) % 2_147_483_647) % 100_000)
        .collect();

    // Insertion sort — same algorithm as all three language variants
    for i in 1..n {
        let key = a[i];
        let mut j = i;
        while j > 0 && a[j - 1] > key {
            a[j] = a[j - 1];
            j -= 1;
        }
        a[j] = key;
    }

    let checksum: i64 = a.iter().enumerate().map(|(i, &v)| v * (i as i64 + 1)).sum();
    println!("{checksum}");
}
